Update Mustache to v2.2.1

petetnt opened the pull request:
This PR updates Mustache to version `v2.2.1` from `v0.7`. 

Mostly everything Brackets uses Mustache for works as it used to except for `Mustache.compile` which has been completely removed for over 2 years. This PR also provides a simple polyfill for it for not to break extensions that might still be using it.

Commit:
Fix the misaligned selection extension when creating a folder

sprintr committed:
Set Bash files syntax highlighting

ficristo committed:
Fix for 3296: add deprecation warning for global Mustache

heysupratim committed:
Added line comment toggling only at start of code added #11953

heysupratim committed:
added line toggling like sublime where subsequent comment position is based on first toggle position fixed #11953

heysupratim committed:
uneven seelcted led to wrong toggle of comments , recalucated curssor position for offset lines #11953

heysupratim committed:
lint error of equality - lint is awesome

heysupratim committed:
removed duplicate variable declaration

heysupratim committed:
re wrote line comment test cases

heysupratim committed:
removed lint errors in EditorCommandHandler-test.js

heysupratim committed:
Merge remote-tracking branch 'upstream/master' into lineCommentBugFix

swmitra committed:
Notification on explicit cursor/selection change

swmitra committed:
Merge branch 'master' of https://github.com/adobe/brackets into swmitra/NavigationHistory

swmitra committed:
Brackets navigation between editor context

swmitra committed:
UI for recent files

swmitra committed:
Merge branch 'master' of https://github.com/adobe/brackets into swmitra/NavigationHistory

swmitra committed:
JSLint warning fix

swmitra committed:
Most recently opened files navigation(MROF)
Open the list using File-> Open Recent (Alt+O)
Navigate using keyboard only - Next (Alt+Shift+N), Prev (Alt+Shift+P)

swmitra committed:
Click handler fix.

swmitra committed:
UI corruption fix.

swmitra committed:
Dirty dot UI corruption fix.
Minor refactoring

swmitra committed:
Fix for duplicate overlapping popovers.

swmitra committed:
Handling Left and Right arrow keys in mixed navigation (mouse+keyboard)

swmitra committed:
Remove Mouse workflow

ficristo committed:
jasmine.sh: add some log to say which platform are support

swmitra committed:
Fix for non working Ctrl+Click and Ctrl+Shift+Click for opening same document on targeted panes

Commit:
Update German translation

ficristo commented:
I'm all for use up-to-date libraries, but is there any reason beyond that to update `Mustache`?
Did you look if there is any extension use `Mustache.compile`?
Maybe using something like https://github.com/dangoor/BracketsExtensionGrabber.

ficristo commented on src/thirdparty/CodeMirror:
```diff
@@ -1 +1 @@
-Subproject commit 61ca2db598f3f1d79e837f326bc740e8bc49e62a
+Subproject commit 291b06eb891ff63bc010795bd492e9a672c13566
```
I don't think this change belong in this PR

petetnt replied:
Fixed

petetnt committed:
Update Mustache to v2.2.1

petetnt commented:
@ficristo No much other reason than it's a low hanging fruit (that I had already done when doing #12006) that could provide some performance benefits, especially for ExtensionManager if the parsing is faster (which it should be).

petetnt committed:
Update React to 0.14.5 and Immutable to newest version

dvorapa committed:
Czech translation for v1.6

ficristo committed:
Remove a warning when reloading the app

petetnt committed:
Remove .jsx from Javascript file extensions and create an jsx-mode

petetnt committed:
Wrap indent-key to parentheses in .eslintrc

The key `indent` was missing parentheses in `.eslintrc`, rendering the file invalid JSON. 

This PR fixes that simple error.

swmitra committed:
Null doc fix

abose committed:
Merge pull request #12054 from adobe/versionBump1.7

version bump to 1.7

abose committed:
version bump to 1.7

walfgithub committed:
Updated by ALF automation.

swmitra committed:
MRU List duplication fix when same document is being flipped

ficristo committed:
Update a couple of dependencies in package.json at the root

swmitra committed:
Fix for Hiding of all resizable divs during relaunch after setting no distractions. Should target only bottom-panel marked divs.

petetnt committed:
Fix Flipview focus issues

petetnt committed:
Remove the pane.retainFocusAfterFlip preference

swmitra committed:
Merge pull request #12060 from petetnt/petetnt/BugFixesFor1.6Beta

Fix Flipview focus issues

walfgithub committed:
Updated by ALF automation.

petetnt committed:
Fix FlipView / WorkingSet / SplitViewSameDoc integration issues

heysupratim committed:
added preference for Editor linecommentIndent

nethip committed:
Merge pull request #12055 from adobe/alf_localization_release_new

ALF Automation

heysupratim committed:
added preference handling -preference indentLineComment with default value false for fix #11953

swmitra committed:
Merge pull request #12065 from petetnt/petetnt/BugFixesFor1.6Beta

Fix FlipView / WorkingSet / SplitViewSameDoc integration issues

heysupratim committed:
made test cases for both prefernce set values fix #11953

heysupratim committed:
removed merge conflicts and added comments

swmitra committed:
Prevent same doc flip to avoid document corruption

swmitra committed:
Merge branch 'release' of https://github.com/adobe/brackets into swmitra/BugFixesFor1.6Beta

swmitra committed:
Fix for the focus issue while switching focus between inline editors hosted under split views for the same doc.

petetnt committed:
Update translation for 1.6 release

Commit:
Update SHA

petetnt committed:
Address comments

nethip committed:
Update README.md

As filing an issue with Brackets is now the only way for users to submit translations suggestions (Fr and JP), updated README to reflect that.

swmitra committed:
Fix for same doc drag across panes in working set and same doc flip using pane header.

nethip committed:
Merge pull request #12072 from adobe/swmitra/BugFixesFor1.6Beta

Bug fixes for 1.6 release

Commit:
Update strings.js

Commit:
Update strings.js

Commit:
Update strings.js

jshen212 committed:
Add period on line 741

Update line 741 to be uniform with other comments

nethip committed:
Merge branch 'release' into master

petetnt committed:
Remove dropshadow from flip-view-buttons

walfgithub committed:
Updated by ALF automation.

Commit:
Korean translation update

naradesign committed:
Fixed Korean grammertical rule.

naradesign committed:
Unification into existing Korean terms.

prayagverma committed:
Make license year evergreen

prayagverma committed:
Make src folder files evergreen

prayagverma committed:
Make tasks folder and Gruntfile file license evergreen

prayagverma committed:
Make test folder license evergreen

heysupratim committed:
Merge branch 'master' into lineCommentBugFix

heysupratim committed:
Removed styling issues and removed unwanted comments

Removed inovking the toggling of comment for 2nd time to get original selection back as the line commend indent feature doesnt
give the same bheviour as an editor without line comment indentation

abose committed:
Merge pull request #12092 from pra85/2016

Update year to 2016 in LICENSE and about-dialog.html

abose committed:
Merge pull request #12151 from naradesign/Korean-translation

Korean translation update

abose committed:
Merge pull request #12024 from ficristo/jasmine-sh-log

jasmine.sh: add some log to say which platforms are support

Commit:
Optimize and clean SVG icons

Commit:
Use SVG in image preview backgrounds

Replace PNG grid with SVG one in image preview backgrounds.

Commit:
Replace PNG backgrounds with SVG ones in QuickView

heysupratim committed:
Merge branch 'master' into lineCommentBugFix

heysupratim committed:
seperated the test cases for both preference set and unset and reset preference back to original value

petetnt committed:
Only update stat and clear contents when old stat is newer than current stat. Potentially fixes  #11826

Commit:
Merge pull request #12165 from valtlait/valtlait/img-preview-bg

Use SVG on image preview backgrounds

Commit:
Update CodeMirror

Commit:
Update expected range when pressing { on a selection
codemirror/CodeMirror@c65244d1f5fa9164cbf62e2328e9a46d614ce465 changed
the behavior in this case; the newly inserted braces are no longer
part of the resulting selection.

abose committed:
Merge pull request #12074 from petetnt/petetnt/fi-translation1.6

Update finnish translation for 1.7 release

abose committed:
Merge pull request #12046 from dvorapa/patch-1

Czech translation for v1.6

swmitra committed:
Adding back mouse workflow

swmitra committed:
Merge branch 'master' of https://github.com/adobe/brackets into swmitra/NavigationHistory

swmitra committed:
Fix for lint error

swmitra committed:
Lint error fix

MauricioCarmelo committed:
Few Portuguese update, spelling and punctuation correction

One of the paragraphs was rewritten because it was out of date in relation to the English version. There were also a few spelling and punctuation corrections.

abose committed:
Merge pull request #12076 from adobe/nethip/update_LOC_DOCUMENTATION

Update README.md

abose committed:
Merge pull request #12085 from jshen212/patch-1

Add period on line 741

abose committed:
Merge pull request #12191 from MauricioCarmelo/master

Few Portuguese update, spelling and punctuation correction

heysupratim committed:
Merge branch 'master' into lineCommentBugFix

heysupratim committed:
split the remaining two tests , hopefully this ends the brackets linecommentindent feature- that was a tough one month

petetnt committed:
Only update stat and clear contents when old stat is newer than current stat. Potentially fixes #11826

ficristo committed:
Require PathUtils instead of using the global one

Commit:
Update strings.js

Update Italian string. I correct many errors!

Commit:
Update strings.js

walfgithub committed:
Updated by ALF automation.

petetnt committed:
Bump React to 0.14.7, immutable to 3.7.6

petetnt committed:
Remove redundant findDomNode, align domNode variables

abose committed:
Merge pull request #11954 from borax12/lineCommentBugFix

Line comment bug fix

MiguelCastillo committed:
Merge pull request #12035 from petetnt/petetnt/update-react

Update React to 0.14.7 and Immutable to 3.7.6.

MiguelCastillo commented on src/brackets.js:
```diff
@@ -114,6 +114,23 @@ define(function (require, exports, module) {
+    // DEPRECATED: Mustache.compile has been deprecated since Mustache 0.8,
+    // but for now we use a polyfill for it so as to avoid breaking extensions in the
+    // interim.
+    Object.defineProperty(window.Mustache, "compile", {
```
This is great. I am not sure how prominent `compile` is, but it is good to warn and tag it a deprecated.  @ficristo is right about letting extension authors this change.  @MarcelGerber might be able to give us some guidance on this. :)

MarcelGerber commented:
I had a look at updating Mustache some months ago as well, but when I did it, I faced some display issues in the Extension Manager.
So we should definitely look out for regressions.

I may get to have a look soon (at least I hope so).

MiguelCastillo committed:
Merge pull request #12053 from petetnt/patch-1

Wrap indent-key to parentheses in .eslintrc

Commit:
Refactoring QuickOpen extensions, separating reused code into a new file.

abose committed:
fix error on live preview server creation. Else node crashes if the port is already in use.

Commit:
Show an error message when trying to rename a file outside of the project

ficristo committed:
Rename .eslintrc to .eslintrc.json

ficristo committed:
Add gyp and gypi file extensions to python language

abose committed:
Merge pull request #12175 from petetnt/petetnt/ctrl-z-hotfix

[HOTFIX] Only update stat and clear contents when old stat is newer than current stat.

abose committed:
Merge pull request #12237 from ficristo/eslint-rename

Rename .eslintrc to .eslintrc.json

abose committed:
Merge pull request #12238 from ficristo/language-gyp

Add gyp and gypi file extensions to python language

abose committed:
Merge pull request #12195 from petetnt/petetnt/ctrl-z-hotfix-release

[HOTFIX] Only update stat and clear contents when old stat is newer than current stat.

Commit:
added one more shortcut ctrl+shift+1 for toggle panels which will be preferred

swmitra committed:
Merge pull request #12243 from adobe/abose/NoDistractionShortcut

Fix12131: release refresh 1.6: added one more shortcut ctrl+shift+1  to toggle panels

swmitra committed:
Targeting PR#12145 to release

abose committed:
Merge pull request #12246 from adobe/swmitra/ReleaseFixBranch

Targeting PR#12145 to release

abose committed:
Merge pull request #12068 from adobe/marcel/german-translation

Update German translation

AllThingsSmitty committed:
Linted CSS

I went through the sample CSS for the “Getting Started” pages,
reordered rulesets properties to an alpha order, removed properties not
needed (based on last 2 versions of major browsers and IE8+) and
removed unnecessary property values (e.g., em units on line-height).

nethip committed:
Merge pull request #12138 from adobe/alf_localization_release_new

ALF Automation

abose committed:
Update README.md

Adding admin email and instructions to join Slack channel.

busykai committed:
Add @mfatekho to CLA exceptions.

@mfatekho is Marat Fatekhov. He is covered by Adobe/Intel CCLA.

nbfontana committed:
Updated First Steps in PT-BR

petetnt committed:
Add "gradle" to Groovy file extensions

Gradle is an open source build automation system that builds upon the concepts of Apache Ant and Apache Maven and introduces a Groovy-based domain-specific language (DSL) instead of the XML form used by Apache Maven of declaring the project configuration.

Fixes #12332

AllThingsSmitty committed:
Reverting to prior font-weight

busykai committed:
Merge pull request #12321 from adobe/kai/cla-addition

Add @mfatekho to CLA exceptions.

abose committed:
Merge pull request #12333 from petetnt/patch-1

Add "gradle" to Groovy file extensions

Commit:
changing to switch case

abose committed:
Merge pull request #12234 from adobe/marcel/rename-outside-project-fail

Show an error message when trying to rename a file outside of the project

abose committed:
Merge pull request #12230 from adobe/abose/nodeSockErr

node crash fix: error on live preview server creation.

abose committed:
Merge pull request #12081 from Denisov21/patch-40

Italian strings.js for 1.7

nbfontana committed:
Grammar corrections

nbfontana committed:
The "o Brackets" commit

abose committed:
Merge pull request #12164 from valtlait/valtlait/svg-images

Optimize and clean up SVG icons

nbfontana committed:
Grammar corrections II

abose committed:
Merge pull request #12334 from nbfontana/samples-pt-br

Updated first steps in portuguese language (pt-br)

abose committed:
Merge pull request #12250 from AllThingsSmitty/master

Linted CSS

abose committed:
Merge pull request #12124 from petetnt/petetnt/flip-view-shadow

Remove dropshadow from flip-view-buttons

Commit:
Update index.html

Removed the word 'And' in Samples/root/index.html to make the sentence correct.

MiguelCastillo committed:
Merge pull request #12344 from JoshLWScott/patch-1

Update index.html

MiguelCastillo committed:
Merge pull request #12177 from adobe/marcel/cm-update

Update CodeMirror

mfatekho committed:
Fix of not picking file modifications.

Inapproriate use of win api in fs_even_node module. In fsevents_win.cpp there are a thread that monitors file system events queue and post messages to uv library. Proposed solution is to use extra bool variable in THREADINFO structure to control thread exit state

Commit:
Added Bulgarian translation.

mfatekho committed:
Innapropriate copyright fixed.

Commit:
Update Polish translation (checked and tested)

nethip committed:
With latest CEF, quick open is not showing the results as opacity was set to 0. So removed opacity for .quick-search-container

Commit:
Change GOTO_DEFINITION string

abose committed:
Merge pull request #12372 from jurkian/master

Update Polish translation (checked and tested)

swmitra committed:
ImagePreview URL protocol validation

swmitra committed:
Handling case insensitive lookup of protocol

mfatekho committed:
Minor refactoring

swmitra committed:
Homepage url validation for extension listing

mfatekho committed:
Indention fixed

Commit:
Added unit testing for issue #8208

Commit:
Merge remote-tracking branch 'upstream/master' into selection-extension-create-folder

Commit:
Merge pull request #10402 from MarcelGerber/selection-extension-create-folder

Fix the misaligned selection extension when creating a folder

Commit:
Merge pull request #11616 from ficristo/fix-3296

Fix for #3296: add deprecation warning for global Mustache

Commit:
Merge pull request #12048 from ficristo/fix-warning-less

Remove a warning when reloading the app

Commit:
Merge pull request #12203 from ficristo/require-path-utils

Require PathUtils instead of using the global one

Commit:
Fix CSSUtils detection of @import urls

Commit:
Fixed indentation, copyright notice, and JsDoc

nethip committed:
Merge pull request #12393 from adobe/marcel/cssutils-fix-import-url

Fix CSSUtils detection of @import urls

Commit:
In CSSUtils, detect when in an @ rule correctly

swmitra committed:
Merge pull request #12401 from adobe/release

Pull interim(after 1.6) fixes from release

abose committed:
Merge pull request #12357 from lyubomirv/bg

Added Bulgarian translation.

bchintx committed:
Merge pull request #12353 from mfatekho/fix-win-filewatchers

Fix of not picking file modifications.

swmitra committed:
Fix for documents refCount issue

swmitra committed:
Missing length

nethip committed:
Merge pull request #11558 from sprintr/bash-startup-files-highlight

Set Bash files syntax highlighting

nethip committed:
Merge pull request #12386 from adobe/swmitra/imagePreview

ImagePreview URL protocol validation

nethip committed:
Merge branch 'master' into prashant/cef-upgrade-latest

nethip committed:
Reverting the windows scroll fix put in by Marcel Gerber via https://github.com/adobe/brackets/pull/10930/

swmitra committed:
Merge branch 'master' of https://github.com/adobe/brackets into swmitra/NavigationHistory

Commit:
Fix animation mixin

swmitra committed:
Add split view and override shorcuts

swmitra committed:
Merge branch 'master' of https://github.com/adobe/brackets into swmitra/HomePageEntryValidation

# Conflicts:
#	src/extensibility/ExtensionManagerView.js

swmitra committed:
Update fsevents_win.cpp

swmitra committed:
Update fsevents_win.cpp

swmitra committed:
Remove unwanted changes

swmitra committed:
Create detached link element only once

nethip committed:
Merge pull request #12389 from adobe/swmitra/HomePageEntryValidation

Homepage url validation for extension listing

swmitra committed:
Merge pull request #12415 from adobe/prashant/cef-upgrade-latest

[CEF 2623] Upgrade to latest CEF

Commit:
Merge pull request #12397 from adobe/marcel/cssutils-detect-import

In CSSUtils, detect when in an @ rule correctly

MiguelCastillo committed:
Merge pull request #12052 from petetnt/petetnt/jsx-mode

Remove .jsx from Javascript file extensions and create an jsx-mode

ficristo committed:
Fixed htmlmixed highlighting. Add text/{babel,jsx} to the jsx mode

MiguelCastillo committed:
Merge pull request #12422 from ficristo/htmlmixed-modes

Fixed htmlmixed highlighting.

swmitra committed:
Code cleanup, UI fixes and handling split scenario

swmitra committed:
Adding keyboard config making strings translatable

swmitra committed:
Doc ref count fix

nethip committed:
Merge pull request #12390 from PatrickDallarosa/master

Added unit testing for issue #8208

swmitra committed:
Added base name conflict handling like workingset

swmitra committed:
Handle Asych filesystem sync

swmitra committed:
Putting back missing ckeck and add to pane views

swmitra committed:
Added async handling for move next/prev

swmitra committed:
Add Preference to enable/disable mrof navigation

swmitra committed:
Changed pref string

swmitra committed:
Changed command names(replaced '.' with '-')

swmitra committed:
Resetting MROF list on prefs change

nethip committed:
Update Document.js

Changing `editorCount++` to `++editorCount`

nethip committed:
Merge pull request #12405 from adobe/swmitra/RefCountFix

Fix for documents refCount issue

swmitra committed:
Fix for workingset sync and event hldr registration

Commit:
Fix ProjectModel test 'adjusts the selection if a parent folder is renamed'
Test failure caused by a505026d40a4eb0bffc196eda30557c0e6b021bc

Commit:
Fix Extension Manager test failures
Caused by interference between #12389 and #12203.
The former utilizes PathUtils, while the latter adapted PathUtils
to our normal require() workflow.

Commit:
Fix CSS Code Hints test case 'should show color swatches for background-color'
Caused by our recent CEF upgrade

nethip committed:
Merge pull request #12437 from adobe/marcel/test-fixes

Some unit test fixes

swmitra committed:
Fix for EditorManager(1) and MainViewManager(4) test failures

swmitra committed:
Addressing review comments

Commit:
Merge pull request #12214 from jacobsone/refactoring

Refactoring QuickOpenHTML, CSS, and JavaScript extensions

Commit:
Fixup on copyright format

Commit:
Merge pull request #12059 from ficristo/update-dependencies

Update a couple of dependencies in package.json at the root

nethip committed:
Merge pull request #12439 from adobe/swmitra/AddPaneTestFix

Fix for EditorManager(1) and MainViewManager(4) test failures

nethip committed:
Merge pull request #12012 from adobe/swmitra/NavigationHistory

Open Recent Files

nethip committed:
Bumping version Number to 1.8

nethip committed:
Merge pull request #12446 from adobe/prashant/VersionUpdate

Bumping version Number to 1.8

Commit:
Merge commit 'refs/pull/12034/head' of https://github.com/adobe/brackets into HEAD

Commit:
Update Mustache - Merge conflicts

MarcelGerber commented:
I've now had a look at this PR, and noticed the issue I had before is still present - in the Extension Manager dialog, extension author's names are no longer shown.

(If you need help in resolving the merge conflicts, please say so, I have a working branch local)

Commit:
Fix for Mustache's issue

MarcelGerber commented:
My latest changes are up at `MarcelGerber:update-mustache` (see https://github.com/adobe/brackets/compare/8093791728cbff6f7734fbad41882481c82087f9...MarcelGerber:update-mustache for a changeset against current master).

I've also filed the issue with Mustache: https://github.com/janl/mustache.js/issues/566

And, to complete, I've looked for extensions that use `Mustache.compile` and found these:
* brackets-cuwire
* brackets-wallpaper
* genuitec.team-extensions (for some reason, they have a copy of `ExtensionManagerView`?)
* shreder *uses `Mustache.compilePartial`*, for which we don't have a polyfill!

petetnt commented:
Blarg, screwed rebasing this PR with your changes. @MarcelGerber feel free to open a new PR with your branch. Really great work finding the issues as always, I'll check if there's a simple way to polyfill the the `compilePartial` method.
